src/utils.py

```diff
@@ -0,0 +1,36 @@
+from os import getenv
+from typing import Optional, Type, TypeVar
+
+T = TypeVar("T")
+
+
+def setting(name: str, default: Optional[T] = None, *, required: bool = False, _type: Type[T] = str) -> T:
```

Comment:
I was just thinking, if `default` and `required` are mutually exclusive, could this be simplified by considering a variable to be required if a default isn't provided.

Reply:
Also, it would make the function more general to replace the `_type: Type[T] = str` argument with something like `converter: Callable[[str], T] = str`.

Although getting the type checking to work for this is a bit weird, it turns out. After some trial and error, I found this works:
```py
T = TypeVar("T")
Converter = Callable[[str], T]

@overload
def setting(name: str, default: Optional[str] = None) -> str: ...

@overload
def setting(name: str, default: Optional[T] = None, *, converter: Converter[T]) -> T: ...

def setting(name, default=None, *, converter=str):
    ...
```
Or alternatively:
```py
@overload
def setting(name: str, default: str = "") -> str: ...

@overload
def setting(name: str, default: str = "", *, converter: Converter[T]) -> T: ...

def setting(name, default="", *, converter=str):
    ...
```

Reply:
These changes are totally optional btw. Up to you.

Reply:
This makes a lot of sense. I would go with your suggestion for
```py
@overload
def setting(name: str, default: str = "", *, converter: Converter[T]) -> T: ...
```

Although if we are having the `required` param left out as the default param makes up for it. I would do this
```py
@overload
def setting(name: str, default: Optional[str] = None, *, converter: Converter[T]) -> T: ...
```
to make the idea that if None is passed it is required, rather than an empty string.